### DISCUSSION TITLE
Doesn't build on Mac OS

### DISCUSSION THREAD

**Discussion by Saddamus** [rustdesk/rustdesk#5969] [Q&A]
Hello, 
Im trying to make a custom build on mac, but it ends up with error below.
The same issue happens if i checkout the original code.
System version: 12.7

`error[E0432]: unresolved imports `apple_sys::IOKit::kIOPMAssertionLevelOn`, `apple_sys::IOKit::kIOReturnSuccess`, `apple_sys::IOKit::CFStringRef`, `apple_sys::IOKit::IOPMAssertionCreateWithName`, `apple_sys::IOKit::IOPMAssertionRelease`
  --> /Users/adam/.cargo/git/checkouts/keepawake-rs-b579583c979e8f5b/ac395ef/src/sys/macos.rs:9:5
   |
9  |     kIOPMAssertionLevelOn, kIOReturnSuccess, CFStringRef, IOPMAssertionCreateWithName,
   |     ^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^  ^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `IOPMAssertionCreateWithName` in `IOKit`
   |     |                      |                 |
   |     |                      |                 no `CFStringRef` in `IOKit`
   |     |                      no `kIOReturnSuccess` in `IOKit`
   |     no `kIOPMAssertionLevelOn` in `IOKit`
10 |     IOPMAssertionRelease,
   |     ^^^^^^^^^^^^^^^^^^^^ no `IOPMAssertionRelease` in `IOKit`
   |
   = help: consider importing this type alias instead:
           core_foundation::string::CFStringRef

For more information about this error, try `rustc --explain E0432`.
error: could not compile `keepawake` (lib) due to previous error`

**Comment by blacktoby** [comment]
For everyone struggling with this error: Using the right Version of rust-c is the key. I installed "stable" with [rustup](https://sourabhbajaj.com/mac-setup/Rust/) and it worked. For me stable was "1.76.0" (rustc --version).
